src/test/java/org/broadinstitute/consent/http/resources/DatasetResourceTest.java

```diff
@@ -314,12 +307,39 @@ void testPatchByDatasetUpdate_patchableNoName() {
     when(datasetRegistrationService.patchDataset(any(), any(), any())).thenReturn(dataset);
     when(authUser.getEmail()).thenReturn("test@test.com");
     when(userService.findUserByEmail("test@test.com")).thenReturn(user);
-    when(user.getUserId()).thenReturn(RandomUtils.nextInt(1, 100));
+    when(user.getUserId()).thenReturn(randomInt(1, 100));
+    dataset.setCreateUserId(user.getUserId());
+
+    initResource();
+    try (Response response = resource.patchByDatasetUpdate(authUser, dataset.getDatasetId(),
+        gson.toJson(patch))) {
+      assertEquals(HttpStatusCodes.STATUS_CODE_OK, response.getStatus());
+    }
+  }
+
+  @Test
+  void testPatchByDatasetUpdate_invokeIndexUpdate() throws Exception {
+    Gson gson = GsonUtil.buildGson();
+
+    Dataset dataset = new Dataset();
+    dataset.setDatasetId(randomInt(1, 100));
+    dataset.setName(randomAlphabetic(10));
+    dataset.setIndexedDate(new Date());
+    when(datasetService.findDatasetById(any())).thenReturn(dataset);
+    String newName = randomAlphabetic(20);
+    DatasetPatch patch = new DatasetPatch(newName, List.of());
+
+    when(datasetRegistrationService.patchDataset(any(), any(), any())).thenReturn(dataset);
+    when(authUser.getEmail()).thenReturn("test@test.com");
+    when(userService.findUserByEmail("test@test.com")).thenReturn(user);
+    when(user.getUserId()).thenReturn(randomInt(1, 100));
     dataset.setCreateUserId(user.getUserId());
+    when(elasticSearchService.indexDataset(dataset, user)).thenReturn(mockResponse);
 
     initResource();
     try (Response response = resource.patchByDatasetUpdate(authUser, dataset.getDatasetId(),
         gson.toJson(patch))) {
+      verify(elasticSearchService, times(1)).indexDataset(dataset, user);
```

Comment:
`times(1)` is the default so it can be omitted.

In this case the `verify()` call isn't necessary since you have a `when()` and strict stubbing is enabled. If you wanted to, you could remove the `when()` and leave the `verify()`, which would have the same effect.

Reply:
> `times(1)` is the default so it can be omitted.

This works, thank you!

> In this case the `verify()` call isn't necessary since you have a `when()` and strict stubbing is enabled. If you wanted to, you could remove the `when()` and leave the `verify()`, which would have the same effect.

Removing `when` and leaving `verify` fails, I do need both to ensure that it is called.